client/src/components/Endpoints/getIcon.tsx

```diff
@@ -0,0 +1,143 @@
+import React from 'react';
+import { Plugin, GPTIcon, AnthropicIcon } from '~/components/svg';
+import { useAuthContext } from '~/hooks';
+import { cn } from '~/utils';
+import { IconProps } from '~/common';
+
+const getIcon: React.FC<IconProps> = (props) => {
+  const { size = 30, isCreatedByUser, button, model = true } = props;
+  // eslint-disable-next-line react-hooks/rules-of-hooks
+  const { user } = useAuthContext();
+
+  if (isCreatedByUser) {
+    const username = user?.name || 'User';
+
+    return (
+      <div
+        title={username}
+        style={{
+          width: size,
+          height: size,
+        }}
+        className={`relative flex items-center justify-center ${props.className || ''}`}
+      >
+        <img
+          className="rounded-sm"
+          src={
+            user?.avatar ||
+            `https://api.dicebear.com/6.x/initials/svg?seed=${username}&fontFamily=Verdana&fontSize=36`
+          }
+          alt="avatar"
+        />
+      </div>
+    );
+  } else {
+    const { endpoint, error } = props;
+    let icon, bg, name;
+
+    switch (endpoint) {
```

Comment:
switch/case should never be used, you won't find it in this project: https://javascript.plainenglish.io/why-you-should-avoid-switch-statements-in-javascript-ce07d5b60da4